### ISSUE TITLE
Unable to curry a function that uses `func_get_args`

### ISSUE THREAD

**Issue by webNeat** [matteosister/php-curry#3]
Hello, I am having an issue when currying a function that uses `func_get_args`. 
Here is the failing test case:

``` php
public function testCurryFunctionUsingFuncGetArgs() {
    $fn = function() {
        return implode(', ', func_get_args());
    };

    $curried = F\curry($fn);

    $this->assertEquals('foo, bar', $fn('foo', 'bar'));
    $this->assertEquals('foo, bar', $curried('foo', 'bar'));
}
```

The first assertion goes fine while the second is failling.

Thanks

**Comment by matteosister** [comment]
@webNeat should we close this?

**Comment by webNeat** [comment]
Yes, this is solved 7da718312e13860778fa6f76330f0dd1e4eb2a9d
